colobot-base/src/script/scriptfunc.cpp

```diff
@@ -223,15 +223,16 @@ CBotTypResult CScriptFunctions::cPlayMusic(CBotVar* &var, void* user)
 
 bool CScriptFunctions::rPlayMusic(CBotVar* var, CBotVar* result, int& exception, void* user)
 {
-    std::string filename;
-    std::string cbs;
-    bool repeat;
-
-    cbs = var->GetValString();
-    filename = std::string(cbs);
+    std::filesystem::path filename;
+    try {
+        std::filesystem::path filename = StrUtils::ToPath(var->GetValString());
+    } catch(...) {
+        exception = ERR_UNKNOWN;
```

Comment:
It look like 'filename' variable is re-declared inside the try block and shadows the one before try.

Maybe ```exception = CBotErrFileOpen```("Can't open file") is better feedback than "Internal error - tell the developers".

Reply:
Oops. Thanks. Fixed